File: onebot/src/main/java/cn/evole/onebot/sdk/util/json/GroupFilesAdapter.java
package cn.evole.onebot.sdk.util.json;

import cn.evole.onebot.sdk.response.group.GroupFilesResp;
import cn.evole.onebot.sdk.util.JsonHelper;
import com.google.gson.*;

import java.lang.reflect.Type;

public class GroupFilesAdapter {
    public static class Files extends JsonHelper implements JsonDeserializer<GroupFilesResp.Files> {
        public GroupFilesResp.Files deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) {
            JsonObject obj = json.getAsJsonObject();
            GroupFilesResp.Files files = new GroupFilesResp.Files();
            files.fileId = obj.get("file_id").getAsString();
            files.fileName = obj.get("file_name").getAsString();
            files.busid = obj.get("busid").getAsInt();
            files.fileSize = obj.get("file_size").getAsLong();
            files.uploadTime = obj.get("upload_time").getAsLong();
            files.deadTime = obj.get("dead_time").getAsLong();
            files.modifyTime = obj.get("modify_time").getAsLong();
            files.downloadTimes = obj.get("download_times").getAsInt();
            files.uploader = obj.get("uploader").getAsLong();
            files.uploaderName = obj.get("uploader_name").getAsString();
            files.md5 = ignorable(obj, "md5", null);
            files.sha1 = ignorable(obj, "sha1", null);
            files.sha3 = ignorable(obj, "sha3", null);
            return files;
        }
    }
}

File: onebot/src/main/kotlin/sdk/util/json/GroupFilesAdapter.kt
package cn.evolvefield.onebot.sdk.util.json

import cn.evolvefield.onebot.sdk.response.group.GroupFilesResp
import cn.evolvefield.onebot.sdk.util.*
import com.google.gson.JsonDeserializationContext
import com.google.gson.JsonDeserializer
import com.google.gson.JsonElement
import java.lang.reflect.Type

class GroupFilesAdapter {
    class Files : JsonDeserializer<GroupFilesResp.Files> {
        override fun deserialize(
            json: JsonElement,
            typeOfT: Type,
            context: JsonDeserializationContext
        ): GroupFilesResp.Files {
            val obj = json.asJsonObject
            return GroupFilesResp.Files().apply {
                fileId = obj.string("file_id")
                fileName = obj.string("file_name")
                busid = obj.int("busid")
                fileSize = obj.long("file_size")
                uploadTime = obj.long("upload_time")
                deadTime = obj.long("dead_time")
                modifyTime = obj.long("modify_time")
                downloadTimes = obj.int("download_times")
                uploader = obj.long("uploader")
                uploaderName = obj.string("uploader_name")
                md5 = obj.nullableString("md5", null)
                sha1 = obj.nullableString("sha1", null)
                sha3 = obj.nullableString("sha3", null)
            }
        }
    }
}
